pkg/payload/precondition/precondition.go

```diff
@@ -71,25 +72,37 @@ func Summarize(errs []error, force bool) (bool, error) {
 		return false, nil
 	}
 	var msgs []string
+	var isWarning bool
 	for _, e := range errs {
 		if pferr, ok := e.(*Error); ok {
 			msgs = append(msgs, fmt.Sprintf("Precondition %q failed because of %q: %v", pferr.Name, pferr.Reason, pferr.Error()))
+			if pferr.NonBlockingWarning {
+				isWarning = true
```

Comment:
This logic isn't quite right, because it treats the existence of a single warning error should not make the overall effect warning.  Instead, the existence of a single blocking alert should make the overall effect blocking.  And also, `force` means we won't block regardless, although it may still be worth splitting informative warnings from more serious "would have been blocking except for force" warnings in that case.

Reply:
The way the current code is written it ignores the warning if there are multiple precondition check failed. Also if there is one error and NonBlockingWarning is not set to true then it will be blocking.

Reply:
https://github.com/openshift/cluster-version-operator/commit/a1e9171fb415e41ff19f32c5741eb3c5e7a16e7d